exchange_machine.lua

```diff
@@ -169,27 +157,26 @@ local function mk_main_fs(p_name, new_item, err_str, success)
 		fs = fs .. "label[0,3;Success!]"
 	end
 
-	fs = fs .. table_from_results(results, 0, 4, 8, 5, selected_def)
-
-	return fs
+	return "size[8,9]" .. fs .. table_from_results(results, 0, 4, 8, 5, selected_def)
 end
 
 
-local function show_main(p_name, new_item, err_str, success)
-	minetest.show_formspec(p_name, main_form, mk_main_fs(p_name, new_item, err_str, success))
+local function show_main(p_name, ...)
+	minetest.show_formspec(p_name, main_form, mk_main_fs(p_name, ...))
 end
 
```

Comment:
ok, l changed it back.
ln my opinion using varargs dots shows that the params aren't directly used by the function but by some function called in it.